x-pack/plugins/security_solution/public/detections/components/rules/related_integrations/utils.tsx

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiLink } from '@elastic/eui';
+import { capitalize } from 'lodash';
+import React from 'react';
+import semver from 'semver';
+import {
+  InstalledIntegration,
+  InstalledIntegrationArray,
+  RelatedIntegration,
+  RelatedIntegrationArray,
+} from '../../../../../common/detection_engine/schemas/common';
+
+/**
+ * Returns an `EuiLink` that will link to a given package/integration/version page within fleet
+ *
+ * @param integration IntegrationDetails describing a package/integrations installed state
+ * @param basePath kbn basepath for composing the fleet URL
+ */
+export const getIntegrationLink = (integration: IntegrationDetails, basePath: string) => {
+  const packageName = integration.package_name;
+  const integrationName = integration.integration_name;
+  const integrationTitle = integration.integration_title ?? integration.package_title;
+  const version = integration.version_satisfied
+    ? integration.package_version
+    : integration.target_version;
+
+  const integrationURL =
+    version !== ''
+      ? `${basePath}/app/integrations/detail/${packageName}-${version}/overview${
+          integrationName ? `?integration=${integrationName}` : ''
+        }`
+      : `${basePath}/app/integrations/detail/${packageName}`;
+  return (
+    <EuiLink href={integrationURL} target="_blank" data-test-subj={'integrationLink'}>
+      {integrationTitle}
+    </EuiLink>
+  );
+};
+
+export interface IntegrationDetails extends InstalledIntegration {
+  target_version: string;
+  version_satisfied: boolean;
+  is_installed: boolean;
+}
+
+/**
+ * Given an array of integrations and an array of installed integrations this will return an
+ * array of integrations augmented with install details like targetVersion, and `version_satisfied`
+ * has
+ * @param integrations
+ * @param installedIntegrations
+ */
+export const getInstalledRelatedIntegrations = (
```

Comment:
Nit: `calculateIntegrationDetails`

Reply:
Ended up being resolved in https://github.com/elastic/kibana/pull/134299